nats/js/kv.py

```
@@ -359,6 +359,48 @@ async def keys(self, **kwargs) -> List[str]:
 
         return keys
 
+    async def keys_with_filters(self, filters: List[str] = None, **kwargs) -> List[str]:
+        """
+        Returns a filtered list of keys in the bucket.
+        If no filters are provided, all keys are returned.
+        """
+        watcher = await self.watchall(
+            ignore_deletes=True,
+            meta_only=True,
+        )
+
+        keys = []
+
+        # Check consumer info and make sure filters are applied correctly
+        try:
+            consumer_info = await watcher.sub.consumer_info()
+            if consumer_info and filters:
+                # If NATS server < 2.10, filters might be ignored.
+                if consumer_info.config.filter_subject != ">":
+                    print("Warning: Server may ignore filters if version is < 2.10.")
+        except Exception as e:
+            print(f"Error fetching consumer info: {e}")
```

should we raise the error here instead?